From 79433632132431c5a4dd372cfb48ec87a1e88f04 Mon Sep 17 00:00:00 2001
From: Colin Patrick Mccabe <cmccabe@cloudera.com>
Date: Tue, 29 Mar 2016 17:13:52 -0700
Subject: [PATCH 1408/2748] HADOOP-12972. Lz4Compressor#getLibraryName returns
 the wrong version number (cmccabe)

(cherry picked from commit e3d15a2e236f93325c67f9e52fb16f645c44238a)
(cherry picked from commit 9df2e54160d3d959138857757d2d64d595b54093)

Change-Id: I39f6419cf74a0048aafb8a21c3015a0e47c3466d
---
 .../apache/hadoop/io/compress/lz4/Lz4Compressor.c  |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/io/compress/lz4/Lz4Compressor.c b/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/io/compress/lz4/Lz4Compressor.c
index 9f14312..8e8777b 100644
--- a/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/io/compress/lz4/Lz4Compressor.c
+++ b/hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/io/compress/lz4/Lz4Compressor.c
@@ -95,7 +95,9 @@ JNIEXPORT jstring JNICALL
 Java_org_apache_hadoop_io_compress_lz4_Lz4Compressor_getLibraryName(
  JNIEnv *env, jclass class
  ) {
-  return (*env)->NewStringUTF(env, "revision:99");
+  char version_buf[128];
+  snprintf(version_buf, sizeof(version_buf), "revision:%d", LZ4_versionNumber());
+  return (*env)->NewStringUTF(env, version_buf);
 }
 
 JNIEXPORT jint JNICALL Java_org_apache_hadoop_io_compress_lz4_Lz4Compressor_compressBytesDirectHC
-- 
1.7.9.5

